src/terminal.go

```diff
@@ -1758,6 +1759,46 @@ func (t *Terminal) executeCommand(template string, forcePlus bool, background bo
 	cleanTemporaryFiles()
 }
 
+func (t *Terminal) executeChangeQuery(template string) {
+	valid, list := t.buildPlusList(template, false)
+	if !valid {
+		return
+	}
+	command := t.replacePlaceholder(template, false, string(t.input), list)
+	cmd := util.ExecCommand(command, false)
+	newQuery := []rune{}
+	t.executing.Set(true)
+	cmd.Stdin = tui.TtyIn()
+	out, _ := cmd.StdoutPipe()
+	t.tui.Pause(false)
+	cmd.Stderr = cmd.Stdout
+	reader := bufio.NewReader(out)
+	err := cmd.Start()
+	if err != nil {
+		newQuery = append(newQuery, []rune(err.Error())...)
+	} else {
+		for {
+			line, err := reader.ReadString('\n')
+			newQuery = append(newQuery, []rune(line)...)
+			if err != nil {
+				break
+			}
+		}
+	}
+	err = cmd.Wait()
+	if err != nil {
+		newQuery = append(newQuery, []rune(err.Error())...)
+	}
+	if newQuery[len(newQuery)-1] == rune('\n') {
+		newQuery = newQuery[:len(newQuery)-1]
+	}
```

Comment:
I was torn on adding this, but decided to put it in.

While testing using `sed 's/^/_/' <<< {q}` as the command, I found that the query was getting a space at the end. It was the newline being added by `sed` being converted somewhere after being set in here.

A lot of commands do some weird stuff to figure out whether or not to include a final newline, and trying to fix it in the command is often clunky at best. So I added the lines in here to do it. I figure a trailing newline is almost always meaningless as far as a query goes. And since it was being turned into a space anyway, it's almost always incorrect to include. Usually, I'd stay away from automatic manipulation like this, but in this case, I think it's warranted. It's much more likely that a user would be confused without it than with it.

Reply:
This also needs to trim `\r\n`.

Use case:
Press ctrl+x to update query with the file extension of current item:
- when I run `fzf.exe --delimiter "\." --bind "ctrl-x:change-query(echo {2})"` and execute it on "test.txt" item in fzf, the query is changed to `"txt"\r`. The `\r` is zero-width, but it does break the filtering. (The double quotes break the filtering as well, but these are straightforward to deal with)